ivadomed/loader/utils.py

```diff
@@ -164,6 +230,73 @@ def get_new_subject_split(path_folder, center_test, split_method, random_seed,
     return train_lst, valid_lst, test_lst
 
 
+def get_new_subject_split_new(df, split_method, data_testing, random_seed,
+                          train_frac, test_frac, log_directory, balance, subject_selection=None):
+    """Randomly split dataset between training / validation / testing.
+
+    Randomly split dataset between training / validation / testing\
+        and save it in log_directory + "/split_datasets.joblib".
+
+    Args:
+        df (pd.DataFrame): Dataframe containing all BIDS image files indexed and their metadata.
+        split_method (str): Used to specify on which metadata to split the dataset (eg. "participant_id", "sample_id", etc.)
+        data_testing (dict): Used to specify the data_type and data_value to include in the testing set.
+        random_seed (int): Random seed.
+        train_frac (float): Training dataset proportion, between 0 and 1.
+        test_frac (float): Testing dataset proportionm between 0 and 1.
+        log_directory (str): Output folder.
+        balance (str): Metadata contained in "participants.tsv" file with categorical values. Each category will be
+        evenly distributed in the training, validation and testing datasets.
+        subject_selection (dict): Used to specify a custom subject selection from a dataset.
+
+    Returns:
+        list, list list: Training, validation and testing subjects lists.
+    """
+    if subject_selection is not None:
+        # Verify subject_selection format
+        if not (len(subject_selection["metadata"]) == len(subject_selection["n"]) == len(subject_selection["value"])):
+            raise ValueError("All lists in subject_selection parameter should have the same length.")
+
+        sampled_dfs = []
+        for m, n, v in zip(subject_selection["metadata"], subject_selection["n"], subject_selection["value"]):
```

Comment:
This function needs to change since in the dataframe, one line is not necessarily one participant. If I one 5 subject with `sex` `M` I'll get 3 lines that might be from the same subject but different modalities. This worked on my end:
```
random.seed(random_seed)
        for m, n, v in zip(subject_selection["metadata"], subject_selection["n"], subject_selection["value"]):
            participants = random.sample(df[df[m] == v]['participant_id'].unique().tolist(), n)
            for participant in participants:
                sampled_dfs.append(df[df['participant_id'] == participant])
```

Reply:
Thank you Andreanne for all the testing! I had missed the subject_selection part.
I fixed it as you suggested,.